src/directions/directions_service.ts

```diff
@@ -216,10 +218,8 @@ export class MigrationDirectionsService {
 
           googleSteps.push({
             distance: {
-              // we do not support Google's behavior of using the unit system of the country of origin and so we will use
-              // Amazon Location's default unit system of kilometers if the unit system option is not specified
-              text: convertKilometersToGoogleDistanceText(step.Distance, options),
-              value: step.Distance * KILOMETERS_TO_METERS_CONSTANT, // in meters, multiply km by 1000
+              text: formatDistanceBasedOnUnitSystem(step.Distance, { ...options, unitSystem }),
+              value: step.Distance,
```

Comment:
Can you add a comment saying that for Google's `value` the distance is always in meters? If someone was just reading this code, they might wonder why the `text` field gets converted based on the unit system but not the `value`.